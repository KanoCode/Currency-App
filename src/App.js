import './styles/App.css';
import { Routes, Route } from 'react-router-dom';
import Navigation from './components/Navigation';
import Profile from './pages/profile';

function App() {
  return (
    <div className="App">

      <Routes>
        <Route path="/" element={<Navigation />} />
        <Route path="/profile" element={<Profile />} />
      </Routes>

    </div>
  );
}

export default App;
