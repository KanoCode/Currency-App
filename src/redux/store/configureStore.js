import { combineReducers, createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import StocksReducer from '../reducers/stocksList';
import ProfileReducer from '../reducers/ProfilesReducer';

const reducers = combineReducers({
  stocks: StocksReducer,
  profile: ProfileReducer,
});

const store = createStore(reducers, applyMiddleware(thunk, logger));

export default store;
