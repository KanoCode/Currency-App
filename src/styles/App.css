* {
  padding: 0;
  margin: 0;
}

h1 {
  color: #fff;
}

header {
  border-bottom: solid darkblue 1px;
  color: #fff;
  justify-content: space-between;
}

header > div {
  width: 16%;
  display: flex;
  justify-content: space-between;
}

.stock-card {
  display: flex;
  margin-top: 3%;
  width: 290px;
  justify-content: space-around;
  align-items: center;
  background: linear-gradient(145deg, #d1d1d5, #f8f8fe);
}

.cpny-details {
  display: flex;
  flex-wrap: wrap;
  text-align: center;
  flex-direction: column;
  align-items: center;
  padding: 3%;
}

.stock-card h5 {
  color: grey;
  font-size: 15px;
}

.stock-card span {
  color: red;
  padding: 3px;
}

.container {
  display: grid;
  grid-template-columns: auto;
  justify-items: center;
}

.stock-card img {
  width: 30%;
}

a {
  text-decoration: none;
}

.card {
  background-color: #d5e9e0;
}

.icon {
  align-items: center;
  justify-content: space-around;
  width: 90%;
}

.cpny-logo {
  display: block;
  max-width: 20%;
  height: auto;
}

.btn-links {
  width: 70%;
  justify-content: space-between;
}

.btn-one {
  background-color: inherit;
  color: inherit;
}

.trends {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.trends span {
  color: #0c59ca;
  display: block;
  font-size: 23px;
}
