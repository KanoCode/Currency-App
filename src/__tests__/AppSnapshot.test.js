import React from 'react';
import Renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store/configureStore';
import App from '../App';

test('renders learn react link', () => {
  const tree = Renderer
    .create(
      <BrowserRouter>
        {' '}
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
