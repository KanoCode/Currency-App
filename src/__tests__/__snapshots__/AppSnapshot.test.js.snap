// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Array [
  " ",
  <div
    className="App"
  >
    <header
      className="App-header d-flex align-items-center"
    >
      <h1>
        Stock data
      </h1>
      <div>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
          />
        </svg>
      </div>
    </header>
    <div
      className="trends"
    >
      <h2>
        Trending now
      </h2>
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h1v15h15v1H0V0zm10 3.5a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 .5.5v4a.5.5 0 0 1-1 0V4.9l-3.613 4.417a.5.5 0 0 1-.74.037L7.06 6.767l-3.656 5.027a.5.5 0 0 1-.808-.588l4-5.5a.5.5 0 0 1 .758-.06l2.609 2.61L13.445 4H10.5a.5.5 0 0 1-.5-.5z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      className="carousel slide"
      data-bs-ride="carousel"
      id="carouselExampleControls"
    >
      <div
        className="carousel-inner"
      >
        <div
          className="carousel-item active"
        >
          <div
            className="card p-1 m-2"
          >
            <div
              className="card-body d-flex flex-column align-items-start"
            >
              <div
                className="icon border d-flex "
              >
                <h5
                  className="card-title"
                >
                  TSLA
                </h5>
                <img
                  alt="Card cap"
                  className="rounded cpny-logo"
                  src="https://financialmodelingprep.com/image-stock/TSLA.png"
                />
              </div>
              <p>
                $ 1223
              </p>
              <span>
                +
                0.23
                 
                (beta)
              </span>
              <div
                className="d-flex btn-links"
              >
                 
                <button
                  className="btn border-dark btn-one "
                  type="button"
                >
                  Sell
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5H9z"
                    />
                  </svg>
                </button>
                 
                <button
                  className="btn btn-dark "
                  type="button"
                >
                  buy
                   
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 9h-2v6.59L5.41 4 4 5.41 15.59 17H9v2h10V9z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <div
            className="card p-1 m-2"
          >
            <div
              className="card-body d-flex flex-column align-items-start"
            >
              <div
                className="icon border d-flex "
              >
                <h5
                  className="card-title"
                >
                  CASH
                </h5>
                <img
                  alt="Card cap"
                  className="rounded cpny-logo"
                  src="https://financialmodelingprep.com/image-stock/CASH.png"
                />
              </div>
              <p>
                $ 1223
              </p>
              <span>
                +
                0.23
                 
                (beta)
              </span>
              <div
                className="d-flex btn-links"
              >
                 
                <button
                  className="btn border-dark btn-one "
                  type="button"
                >
                  Sell
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5H9z"
                    />
                  </svg>
                </button>
                 
                <button
                  className="btn btn-dark "
                  type="button"
                >
                  buy
                   
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 9h-2v6.59L5.41 4 4 5.41 15.59 17H9v2h10V9z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <div
            className="card p-1 m-2"
          >
            <div
              className="card-body d-flex flex-column align-items-start"
            >
              <div
                className="icon border d-flex "
              >
                <h5
                  className="card-title"
                >
                  META
                </h5>
                <img
                  alt="Card cap"
                  className="rounded cpny-logo"
                  src="https://financialmodelingprep.com/image-stock/FB.png"
                />
              </div>
              <p>
                $ 1223
              </p>
              <span>
                +
                0.23
                 
                (beta)
              </span>
              <div
                className="d-flex btn-links"
              >
                 
                <button
                  className="btn border-dark btn-one "
                  type="button"
                >
                  Sell
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5H9z"
                    />
                  </svg>
                </button>
                 
                <button
                  className="btn btn-dark "
                  type="button"
                >
                  buy
                   
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19 9h-2v6.59L5.41 4 4 5.41 15.59 17H9v2h10V9z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        className="carousel-control-prev"
        data-bs-slide="prev"
        data-bs-target="#carouselExampleControls"
        type="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="visually-hidden"
        >
          Previous
        </span>
      </button>
      <button
        className="carousel-control-next"
        data-bs-slide="next"
        data-bs-target="#carouselExampleControls"
        type="button"
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="visually-hidden"
        >
          Next
        </span>
      </button>
    </div>
    <div
      className="container"
    >
      <form
        className="p-1 mx-2"
        onSubmit={[Function]}
      >
        <label
          htmlFor="search"
        >
          <input
            className="p-2"
            id="search"
            maxLength="7"
            minLength="1"
            onChange={[Function]}
            placeholder="company search"
          />
          <button
            className="btn btn-primary search"
            type="submit"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </label>
      </form>
    </div>
  </div>,
]
`;
